.github/workflows/ci.yml

```diff
@@ -61,4 +61,4 @@ jobs:
         toolchain: 1.42.0
         components: clippy
     - name: Run tests
-      run: cargo test --verbose --features base64_format,trace,metrics,serialize,binary_propagator,tokio,async-std,serde,http,tonic,reqwest -p opentelemetry -p opentelemetry-jaeger -p opentelemetry-zipkin
+      run: cargo test --verbose --all-features -p opentelemetry -p opentelemetry-jaeger -p opentelemetry-zipkin
```

Comment:
We can't use `--all-features` here because there are some optional features like `surf` support which do not support rust 1.42.0. The msrv tests _should_ be failing for this pr?

Reply:
May need to add `override: true`

Reply:
What do you mean by adding `override: true`? What would that do?

I did verify that "surf" is getting enabled and the test passes, so let's just leave it like this for now?

Reply:
Does `$ cargo +1.42.0 test --all-features` work for you locally?

Reply:
(I think without `default` or `override` set to `true` it will not be using 1.42.0 in tests if I'm reading https://github.com/actions-rs/toolchain#inputs correctly)

Reply:
I added those in #320 so should be able to verify by merging master